### PR TITLE
prombench: Added support for `--bench.version` and `--bench.directory` flags.

### PR DESCRIPTION
**Pull request by bwplotka**
See https://github.com/prometheus/proposals/pull/41 for rationales.

Prometheus job got updated with https://github.com/prometheus/prometheus/pull/15682

Signed-off-by: bwplotka <bwplotka@gmail.com>

### CODE REVIEW

File: prombench/Makefile [hunk 1]
```diff
@@ -37,50 +30,96 @@ cluster_delete:
...
+# prombench GH job on Prometheus repo to call infra CLI with the non-master BENCHMARK_VERSION.
+# In such a case we pull a prombench repository for the given branch or commit version
+# and adjust PROMBENCH_DIR. As a result `make deploy` and `make clean` jobs
+# will apply /manifests/ apply custom manifests or even node pools.
+.PHONE: maybe_pull_custom_version
+maybe_pull_custom_version:
+ifeq (${BENCHMARK_VERSION},master)
+	@echo ">> Using standard benchmark configuration, from the docker image"
+else
+	@echo ">> Git pulling custom benchmark configuration from the ${BENCHMARK_VERSION}"
```
Comment (ArthurSens) [inline prombench/Makefile:52]:
```suggestion
	@echo ">> Git pulling custom benchmark configuration from the ${BENCHMARK_VERSION} branch"
```

nit

File: prombench/Makefile [hunk 2]
```diff
@@ -37,50 +30,96 @@ cluster_delete:
...
+	@$(eval $@_TMP_DIR=$(shell mktemp -d -t "prombench"))
+	cd ${$@_TMP_DIR} && git clone ${PROMBENCH_GIT_REPOSITORY}
+ifeq ($(subst @,,${BENCHMARK_VERSION}),${BENCHMARK_VERSION})
```
Comment (ArthurSens) [inline prombench/Makefile:55]: We rely on the presence of the `@` character to tell if we're receiving a branch or a commit SHA? I do not understand this part 🤔

Comment (ArthurSens) [reply prombench/Makefile:55]: Ok I found my own answer: https://github.com/prometheus/proposals/pull/41/files#diff-b472a523ad2ffdc696851b117f2ee2808d94166c886bf6f090294dc2e728e5e7R106

Indeed you mention the usage of `@` character

File: prombench/README.md [hunk 3]
```diff
@@ -54,3 +66,6 @@ Build the Docker image with:
 docker build -t prominfra/prombench:master .
 ```
 
+
+
+
```
Comment (ArthurSens) [inline prombench/README.md]: Accidently added?

File: prombench/manifests/prombench/README.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+    > WARN: When customizing this directory, don't change `1a_namespace.yaml` or `1c_cluster-role-binding.yaml` filenames as they are used for cleanup routine. Or, if you change it, know what you're doing in relation to [`make clean` job](../../Makefile).
+
+3. Push changes to the new branch.
+4. From the Prometheus PR comment, call prombench as `/prombench <release> --bench.version=benchmark/scenario1` or `/prombench <release> --bench.version=@<relevant commit SHA from the benchmark/scenario1>` to use configuration files from this custom branch.
+
+Other details:
+
+* Other custom branch modifications other than to this directory do not affect prombench (e.g. to infra CLI or makefiles).
+* `--bench.version` is designed for a short-term or even one-off benchmark scenario configurations. It's not designed for long-term, well maintained scenarios. For the latter reason we can later e.g. maintain multiple `manifests/prombench` directories and introduce a new `--bench.directory` flag.
```
Comment (ArthurSens) [inline prombench/manifests/prombench/README.md]: This reads a bit strange to me. You say that we can later introduce a new --bench.directory flag, but this PR already implements this flag, doesn't it?

**Review by ArthurSens — COMMENTED** [review]
I've reviewed only Makefile and manifests/documentation changes and had to stop for some time 😬 

I've left a couple of questions and will hopefully find more time to review the go code later :)

File: prombench/Makefile (hunk 1 above)
Comment (bwplotka) [reply prombench/Makefile:52]: It can also be commit at this point, so skipping - it will be clarified later (:

File: prombench/Makefile (hunk 2 above)
Comment (bwplotka) [reply prombench/Makefile:55]: Correct. This follows docker image reference format. Allows us to never confuse branch and commit (I doubt you can create branches with `@` chars - and even if you can we can block that).

File: prombench/README.md (hunk 3 above)
Comment (bwplotka) [reply prombench/README.md]: yup. Thanks!

File: prombench/manifests/prombench/README.md (hunk 4 above)
Comment (bwplotka) [reply prombench/manifests/prombench/README.md]: yes, I decided to add it eventually so let me update this...

File: prombench/Makefile [hunk 5]
```diff
@@ -37,50 +30,96 @@ cluster_delete:
...
+PROMBENCH_DIR ?= .
+
+# maybe_pull_custom_version allows custom benchmarking as designed in
+# https://github.com/prometheus/proposals/pull/41. It allows calling
+# /prombench <release> --bench.version=<@commit or branch> which will cause
+# prombench GH job on Prometheus repo to call infra CLI with the non-master BENCHMARK_VERSION.
+# In such a case we pull a prombench repository for the given branch or commit version
+# and adjust PROMBENCH_DIR. As a result `make deploy` and `make clean` jobs
+# will apply /manifests/ apply custom manifests or even node pools.
+.PHONE: maybe_pull_custom_version
```
Comment (bwplotka) [inline prombench/Makefile]: LOL, I just noticed ☎️ XD

It should be [`.PHONY:`](https://www.gnu.org/software/make/manual/html_node/Phony-Targets.html) but I copied other commands 🙈 will fix later.

**Comment by bwplotka** [comment]
comment-monitor is deployed and tested in https://github.com/prometheus/prometheus/pull/15487#issuecomment-2553921527

**Comment by bwplotka** [comment]
Hit the limit of what I can test without merging. Merging on green as discussed on Slack with @ArthurSens  https://cloud-native.slack.com/archives/C01AUBA4PFE/p1734606628352259?thread_ts=1734450867.313389&cid=C01AUBA4PFE

**Pull request merged**